### ISSUE TITLE
About Focal loss

### ISSUE THREAD

**Issue by kirin-from-16** [wgcban/ChangeFormer#109]
I'm dealing with a heavy imbalanced dataset of 3 class, with unchange class occupies for 90% of the total training images, but up to 99% of total pixels. For this reason, i choose Focal loss. 

However, in your implementation, the alpha parameter for each class is calculated in a way that make them larger than 1. This goes against the original paper, where the alpha parameters are calculated to be less than 1.
![image](https://github.com/user-attachments/assets/d7b84dcb-93ee-45be-a226-d40d8d5e41b3)

Could you please explain the reason behind this choice? 

Also, the dataloader used to get_alpha was the training set, which applied transform for each run, which lead to different values of alpha for each run. This seem to be a problem.
![image](https://github.com/user-attachments/assets/43f2026b-c39b-4892-a11d-2dc1a8c883be)

The datasets in your paper is also heavily imbalanced. Can you  explain the reason why you choose CrossEntropy instead? Did this cause difficulties in training the model?
